Actions/Packages.json

```diff
@@ -1,3 +1,6 @@
 {
-    "sign": "0.9.1-beta.24123.2"
+    "sign": "0.9.1-beta.24123.2",
+    "Az.Accounts": "1.0.0",
```

Comment:
This is a 5,5-year-old package. Could we not require a higher version than that? Something from within the last year for example.

From what I can see GitHub runners have Az 11.3.1 installed by default. Could we use that as a baseline?  
https://www.powershellgallery.com/packages/Az/11.3.1

Az 11.3.1 requires:
[Az.Accounts](https://www.powershellgallery.com/packages/Az.Accounts/) (>= 2.15.1)
[Az.KeyVault](https://www.powershellgallery.com/packages/Az.KeyVault/) (= 5.2.0)
[Az.Storage](https://www.powershellgallery.com/packages/Az.Storage/) (= 6.1.1)

Reply:
> update them for no reason at all

Not sure this is true. Vulnerabilities is one reason why we'd want to keep these packages up to date. See for example https://github.com/Azure/azure-powershell/issues/23752
https://github.com/Azure/azure-powershell/issues/15035

I'm not suggesting we update to latest-latest but using a relatively recent version I think would be acceptable. Then if people have self-hosted runners that use a 4–5-year-old version of these packages, they'd just have to update the installed software on their runners.

Reply:
Also if the runner was created in November 2023 (one of the vulnerabilities is from December 2023).

This PR is about federated credentials - before we start demanding specific versions of stuff on self-hosted runners, then let's do the following:

1. Add this to the new telemetry module to see what people use - agent version, PowerShell version, Windows version etc.
2. Add this requirement to all vulnerable items - incl. PowerShell, Windows etc. as warnings and not as something that is breaking their builds - maybe something we do in CheckForUpdates. During the builds, we should demand the version we work with.

If people Upgrade AL-Go System Files - they shouldn't be faced with having to update a number of self-hosted runners before their builds run again.

Reply:
The way this file is meant to be read is "AL-Go requires these versions of these packages to run". This isn't true at the moment. Or at least I highly doubt federated credentials are supported in v1.0.0.0 of the Az.Accounts module.

> If people Upgrade AL-Go System Files - they shouldn't be faced with having to update a number of self-hosted runners before their builds run again.

Of course they should! If they want to run on the latest AL-Go version, they need to make sure their runners meet the requirements. Otherwise, we'll be ones paying the tax when dealing with issues because of weird combos of the az modules. Our work is to make them aware _before_ they update AL-Go to the latest version.

Reply:
I don't know which version of Az supports federated credentials - it has been supported in OAuth 2.0 since 2019 (https://darutk.medium.com/oauth-2-0-client-authentication-4b5f929305d4), but I do know that nobody uses it today (except for me in my repositories)

I agree that we should make people aware before they upgrade - meaning that I think we should have a task, which runs in CheckForUpdates (that uses runs-on runner) and a similar task running in build (that is using gitHubRunner runner), which checks that people are using at least the same version of things as the GitHub hosted runners and give warnings if not.

But... - if you guys really feel so strong about this - I will add the version numbers from GitHub hosted runners to packages json and see what happens.

Reply:
You can also include this part in the release notes so that it's documented. E.g. "AL-Go for GitHub requires the following versions of PS modules and will install them if they are not present on the runner" (or something like that).

Reply:
That is a good idea - will do that and re-apply for approval